swap/src/bitcoin/wallet.rs

```diff
@@ -301,11 +305,30 @@ where
         let wallet = self.wallet.lock().await;
         let client = self.client.lock().await;
         let fee_rate = client.estimate_feerate(self.target_block)?;
+        let script = address.script_pubkey();
 
         let mut tx_builder = wallet.build_tx();
-        tx_builder.add_recipient(address.script_pubkey(), amount.as_sat());
+        tx_builder.add_recipient(script.clone(), amount.as_sat());
         tx_builder.fee_rate(fee_rate);
         let (psbt, _details) = tx_builder.finish()?;
+        let mut psbt: PartiallySignedTransaction = psbt;
+
+        // When subscribing to transactions we depend on the relevant script being at
+        // output index 0, thus we ensure the relevant output to be at index `0`.
+        psbt.outputs.sort_by(|a, _| {
+            if a.witness_script == Some(script.clone()) {
```

Comment:
This should allow you to avoid the `.clone()`.

```suggestion
            if a.witness_script.as_ref() == Some(&script) {
```